frontend/src/services/grandmaster-manager/index.ts

```diff
@@ -0,0 +1,100 @@
+import Web3 from "web3";
+import { ManagerError } from "./errors";
+
+interface AccountDetails {
+  accountAddress: string;
+  balance: string;
+  chainId: number;
+}
+
+
+export class GrandMasterManager {
+  private web3Client;
+  constructor() {
+    const win = window as any;
+    this.web3Client = new Web3(win.xdc ? win.xdc : win.ethereum);
+  }
+  
+  /**
+   * * This method will detect XDC-Pay and verify if customer has alraedy loggin.
+   * @returns Account details will be returned if all good, otherwise, relevant error message and type will be returned such as WALLET_NOT_LOGIN
+   */
+  async login(): Promise<AccountDetails | ManagerError> {
+    return {
+      accountAddress: "xdc888c073313b36cf03cf1f739f39443551ff12bbe",
+      balance: "123",
+      chainId: 551
```

Comment:
From the design we need redeemable balance, network id, denom and RPC

Reply:
Also where is `chainId` being used for

Reply:
network id is the chainId. I can help change that term here within this layer
denom is something i have missed. i will add it here but i haven't figure out where/how i can get this value